### PR TITLE
HKG: Car Port for Genesis G80 2024 (2.5T, HDA2)

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**

- [x] added entry to CAR in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: `66eaa6c3b6b2afc6/00000009--3a5199aabe`
- [x] route with stock system: `66eaa6c3b6b2afc6/00000006--1fb9d3a187`
- [x] car harness used (if comma doesn't sell it, put N/A): Harness P
- [x] car architecture: CAN-FD

Superseded https://github.com/commaai/openpilot/pull/32633

### CODE REVIEW

File: opendbc/car/hyundai/interface.py [hunk 1]
```diff
@@ -44,14 +44,15 @@ def _get_params(ret: structs.CarParams, candidate, fingerprint, car_fw, experime
...
-            ret.flags |= HyundaiFlags.CANFD_ALT_GEARS.value
         if candidate not in CANFD_RADAR_SCC_CAR:
           ret.flags |= HyundaiFlags.CANFD_CAMERA_SCC.value
+
+      # ICE cars do not have 0x130; GEARS message on 0x40 or 0x70 instead
+      if 0x130 not in fingerprint[CAN.ECAN]:
+        if 0x40 not in fingerprint[CAN.ECAN]:
+          ret.flags |= HyundaiFlags.CANFD_ALT_GEARS_2.value
+        else:
+          ret.flags |= HyundaiFlags.CANFD_ALT_GEARS.value
```
Comment (jyoung8607) [inline opendbc/car/hyundai/interface.py:55]: Not clear if this was intentional, or for debugging, but would definitely affect others cars.

File: opendbc/car/hyundai/interface.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def _get_params(ret: structs.CarParams, candidate, fingerprint, car_fw, experime
     # FIXME: the Optima Hybrid 2017 uses a different SCC12 checksum
     ret.dashcamOnly = candidate in {CAR.KIA_OPTIMA_H, }
 
-    hda2 = Ecu.adas in [fw.ecu for fw in car_fw]
+    hda2 = True # Ecu.adas in [fw.ecu for fw in car_fw]
```
Comment (jyoung8607) [inline opendbc/car/hyundai/interface.py:28]: Pretty sure this wasn't intentional.

**Review by jyoung8607 — CHANGES_REQUESTED** [review]

File: opendbc/car/hyundai/interface.py (hunk 1 above)
Comment (sunnyhaibin) [reply opendbc/car/hyundai/interface.py:55]: This is intentional: https://github.com/commaai/opendbc/pull/1284

File: opendbc/car/hyundai/interface.py (hunk 2 above)
Comment (sunnyhaibin) [reply opendbc/car/hyundai/interface.py:28]: Opened PR to handle this change: https://github.com/commaai/opendbc/pull/1285

File: opendbc/car/hyundai/values.py [hunk 3]
```diff
@@ -539,6 +539,10 @@ class CAR(Platforms):
     CarSpecs(mass=2060, wheelbase=3.01, steerRatio=16.5),
     flags=HyundaiFlags.LEGACY,
   )
+  GENESIS_G80_2ND_GEN_FL = HyundaiCanFDPlatformConfig(
+    [HyundaiCarDocs("Genesis G80 (2.5T Advanced Trim, with HDA II) 2024", "Highway Driving Assist II", car_parts=CarParts.common([CarHarness.hyundai_p]))],
+    CarSpecs(mass=2060, wheelbase=3.00, steerRatio=11.2),
```
Comment (jyoung8607) [inline opendbc/car/hyundai/values.py:544]: image

```suggestion
    CarSpecs(mass=2060, wheelbase=3.00, steerRatio=14.0),
```

**Review by jyoung8607 — APPROVED** [review]

**Pull request merged**